lib/plugins/personalization.js

```diff
@@ -23,16 +23,31 @@ function configure (config) {
   return this
 }
 
-function fetchUserSegments (audienceToken, successCallback, errorCallback) {
+function fetchUserSegments (tokenOrConfig, successCallback, errorCallback) {
+  var isConfigObject = _.isObject(tokenOrConfig) && !_.isArray(tokenOrConfig)
+  var audienceToken = isConfigObject ? tokenOrConfig.audienceToken : tokenOrConfig
+  var keys = (isConfigObject && tokenOrConfig.keys) || {}
+
   successCallback = successCallback || noop
   errorCallback = errorCallback || noop
 
   invariant(
-    typeof audienceToken === 'string',
-    'audienceToken must be a string; received "' + audienceToken.toString() + '"'
+    typeof audienceToken === 'string' || _.isArray(audienceToken),
+    'audienceToken must be a string or array; received "' + audienceToken.toString() + '"'
   )
 
-  var url = 'https://' + this.client.cdpHost + '/cdp/lookup/collect/segments?token=' + audienceToken
+  invariant(
+    _.isObject(keys),
+    'keys must be an object; received "' + keys + '"'
+  )
+
+  var token = _.isArray(audienceToken) ? audienceToken.join(',') : audienceToken
+  var keyString = _.keys(keys)
+    .map(function (key) {
```

Comment:
can we use `map` if we're supporting IE8?

Reply:
nope :(